src/pycountry/db.py

```diff
@@ -62,31 +64,28 @@ class Subdivision(Data):
     pass
 
 
-def lazy_load(f):
+def lazy_load(f: F) -> F:
     def load_if_needed(self, *args, **kw):
         if not self._is_loaded:
             with self._load_lock:
                 self._load()
         return f(self, *args, **kw)
 
-    return load_if_needed
+    return cast(F, load_if_needed)
 
 
-class Database:
-    data_class: Union[Type, str]
-    root_key: Optional[str] = None
+class Database(Generic[D]):
+    factory: Type[D]
+    indices: Dict[str, Dict[Any, D]]
     no_index: List[str] = []
+    objects: List[D]
+    root_key: Optional[str] = None
```

Comment:
I defined the type of a few of these up here since they all aren't defined in `__init__` and it makes it clearer what they are.